app/models/text_field_type.rb

```diff
@@ -25,14 +28,23 @@ def mapping_field_name
   end
 
   def text_present
-    errors.add(:text, "must be present") if @text.empty?
+    errors.add(:text, 'must be present') if @text.empty?
   end
 
   def text_length
     validator = LengthValidator.new(validations[:length].merge(attributes: [:text]))
     validator.validate_each(self, :text, text)
   end
 
+  def text_unique
+    errors.add(:base, 'Slug Must be unique') unless FieldItem.jsonb_contains(:data, text: text).empty?
+    true
+  end
+
+  def validate_uniqueness?
```

Comment:
Great points! I think we're on the same page for composition vs inheritance - that's why the validations-as-includable-modules approach was suggested. Favoring composition doesn't preclude us from taking a generic approach to validation, however. It shouldn't be too difficult to create a custom Validator class that can take in the JSONB key it needs to act on, but we can save it for V3. Remember: this isn't your everyday application - we want to create a dev toolkit that others can use to create their own tools. We can't assume anything about their use case, and we should favor building for that. In this case, though, the returns for a generic UniquenessValidator are diminishing until we can prove that value - so once the other comments are addressed, this should be good for a merge.